plugins/modules/foreman_subnet.py

```diff
@@ -220,37 +207,34 @@ def main():
         required_one_of=[['cidr', 'mask']],
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    try:
-        # Try to find the Subnet to work on
-        entity = find_subnet(module, name=entity_dict['name'], failsafe=True)
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if 'mask' in entity_dict and 'cidr' not in entity_dict:
-        entity_dict['cidr'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['mask'])).prefixlen
-    elif 'mask' not in entity_dict and 'cidr' in entity_dict:
-        entity_dict['mask'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['cidr'])).netmask
+    # Try to find the Subnet to work on
```

Comment:
I wonder if this comment is really worth it now, especially as the try/except is hidden from this code.

Reply:
:scissors: